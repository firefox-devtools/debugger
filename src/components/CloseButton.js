const React = require("react");
const { DOM: dom, PropTypes } = React;
const Svg = require("./utils/Svg");

require("./CloseButton.css");

function CloseButton({ handleClick, buttonClass }) {
  return dom.div({
    className: buttonClass ? `close-btn-${buttonClass}` : "close-btn",
    onClick: handleClick
  },
    Svg("close"));
}

CloseButton.propTypes = {
  handleClick: PropTypes.func.isRequired
};

module.exports = CloseButton;
