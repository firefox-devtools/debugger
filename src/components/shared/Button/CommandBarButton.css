.command-bar-button {
  appearance: none;
  background: transparent;
  border: none;
  display: inline-block;
  text-align: center;
  position: relative;
  padding: 0px 5px;
  margin-inline-end: 0.3em;
  fill: currentColor;
}

.command-bar-button:focus {
  outline: none;
}

.command-bar-button:disabled {
  opacity: 0.8;
  cursor: default;
}

.command-bar-button:not(.disabled):hover {
  background: var(--theme-toolbar-background-hover);
}

:root.theme-dark .command-bar-button {
  color: var(--theme-body-color);
}

.command-bar-button > * {
  width: 16px;
  height: 16px;
  display: inline-block;
  vertical-align: middle;
}
