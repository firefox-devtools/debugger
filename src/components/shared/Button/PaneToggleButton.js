/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow
import React, { PureComponent } from "react";
import classnames from "classnames";
import Svg from "../Svg";
import { CommandBarButton } from "./";
import "./styles/PaneToggleButton.css";

type Props = {
  collapsed: boolean,
  handleClick: (string, boolean) => void,
  horizontal: boolean,
  position: string
};

class PaneToggleButton extends PureComponent<Props> {
  static defaultProps = {
    horizontal: false
  };

  render() {
    const { position, collapsed, horizontal, handleClick } = this.props;
    const title = !collapsed
      ? L10N.getStr("expandPanes")
      : L10N.getStr("collapsePanes");

    return (
      <CommandBarButton
        className={classnames("toggle-button", position, {
          collapsed,
          vertical: !horizontal
        })}
        onClick={() => handleClick(position, collapsed)}
        title={title}
      >
        <Svg name="togglePanes" />
      </CommandBarButton>
    );
  }
}

export default PaneToggleButton;
