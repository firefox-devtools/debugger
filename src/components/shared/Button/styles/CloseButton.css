/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.close-btn {
  width: 14px;
  height: 14px;
  border: 1px solid transparent;
  border-radius: 2px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0;
}

.close-btn .close {
  mask: url(/images/close.svg) no-repeat;
  mask-size: 100%;
  background-color: var(--theme-comment);
  width: 8px;
  height: 8px;
  padding: 0;
  margin-top: 0;
}

.close-btn .close:hover {
  background-color: var(--theme-comment);
}

.close-btn:hover .img.close {
  background-color: white;
}

.close-btn:hover {
  background-color: var(--theme-selection-background);
}

.close-btn:focus {
  background-color: var(--theme-selection-background);
}

.close-btn:focus .img.close {
  background-color: white;
}

.close-btn.big {
  width: 16px;
  height: 18px;
}

img.close::before {
  width: 100%;
  height: 100%;
  padding: 6px;
}
