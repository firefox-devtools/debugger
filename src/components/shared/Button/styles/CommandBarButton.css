/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.command-bar-button {
  appearance: none;
  background: transparent;
  border: none;
  display: inline-block;
  text-align: center;
  position: relative;
  padding: 0px 5px;
  fill: currentColor;
}

.command-bar-button:disabled {
  opacity: 0.8;
  cursor: default;
}

.command-bar-button:not(.disabled):hover {
  background: var(--theme-toolbar-background-hover);
}

:root.theme-dark .command-bar-button {
  color: var(--theme-body-color);
}

.command-bar-button > * {
  width: 16px;
  height: 16px;
  display: inline-block;
  vertical-align: middle;
}
