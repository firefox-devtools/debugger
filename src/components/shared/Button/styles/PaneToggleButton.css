/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.toggle-button {
  transform: translate(0, 0px);
  transition: transform 0.25s ease-in-out;
  padding: 5px;
  height: inherit;
}

.toggle-button .img {
  fill: var(--theme-comment);
  vertical-align: 0;
}

:root.theme-dark .toggle-button .img {
  fill: var(--theme-comment-alt);
}

.toggle-button.end {
  margin-inline-end: 0px;
  margin-inline-start: auto;
}

.toggle-button.start {
  margin-inline-start: 0px;
}

html:not([dir="rtl"]) .toggle-button.end .img,
html[dir="rtl"] .toggle-button.start .img {
  transform: rotate(180deg);
}

html .toggle-button.end.vertical .img {
  transform: rotate(-90deg);
}

.toggle-button.start.collapsed,
.toggle-button.end.collapsed {
  transform: rotate(180deg);
}
