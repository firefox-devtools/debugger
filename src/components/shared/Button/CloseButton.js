/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow
import React from "react";
import "./styles/CloseButton.css";

type Props = {
  handleClick: Function,
  buttonClass?: string,
  tooltip?: string
};

function CloseButton({ handleClick, buttonClass, tooltip }: Props) {
  return (
    <button
      className={buttonClass ? `close-btn ${buttonClass}` : "close-btn"}
      onClick={handleClick}
      title={tooltip}
    >
      <img className="close" />
    </button>
  );
}

export default CloseButton;
