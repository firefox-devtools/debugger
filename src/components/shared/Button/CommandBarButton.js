/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow
import classnames from "classnames";
import React from "react";

import "./styles/CommandBarButton.css";

type Props = {
  children: React$Element<any>,
  className: string,
  pressed?: boolean
};

export function debugBtn(
  onClick: ?Function,
  type: string,
  className: string,
  tooltip: string,
  disabled: boolean = false,
  ariaPressed: boolean = false
) {
  return (
    <CommandBarButton
      className={classnames(type, className)}
      disabled={disabled}
      key={type}
      onClick={onClick}
      pressed={ariaPressed}
      title={tooltip}
    >
      <img className={type} />
    </CommandBarButton>
  );
}

const CommandBarButton = (props: Props) => {
  const { children, className, pressed = false, ...rest } = props;

  return (
    <button
      aria-pressed={pressed}
      className={classnames("command-bar-button", className)}
      {...rest}
    >
      {children}
    </button>
  );
};

export default CommandBarButton;
