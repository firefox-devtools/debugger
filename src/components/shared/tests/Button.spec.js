import React from "react";
import { shallow } from "enzyme";

import CloseButton from "../Button/Close";
import CommandBarButton from "../Button/CommandBarButton";
import PaneToggleButton from "../Button/PaneToggle";

describe("CloseButton", () => {
  const spy = jest.fn();
  const tooltip = "Close something";
  const wrapper = shallow(
    <CloseButton buttonClass="error" handleClick={spy} tooltip={tooltip} />
  );
  test("renders", () => expect(wrapper).toMatchSnapshot());
  wrapper.find("div").simulate("click");
  test("handleClick is called", () => expect(spy).toHaveBeenCalled());
  test("tooltip is passed through", () =>
    expect(wrapper.find("div").prop("title")).toEqual(tooltip));
});

describe("CommandBarButton", () => {
  test("renders", () => {
    expect(
      shallow(
        <CommandBarButton className="resume" pressed={false}>
          <img />
        </CommandBarButton>
      )
    ).toMatchSnapshot();
  });
});

describe("PaneToggleButton", () => {
  const spy = jest.fn();
  const wrapper = shallow(
    <PaneToggleButton collapsed={false} handleClick={spy} position="start" />
  );
  test("renders start", () => expect(wrapper).toMatchSnapshot());
  test("renders start horizontal", () => {
    wrapper.setProps({ horizontal: true });
    expect(wrapper).toMatchSnapshot();
  });
  test("renders start collapsed", () => {
    wrapper.setProps({ collapsed: true, horizontal: false });
    expect(wrapper).toMatchSnapshot();
  });
  test("renders end", () => {
    wrapper.setProps({ position: "end" });
    expect(wrapper).toMatchSnapshot();
  });
  test("renders end horizontal", () => {
    wrapper.setProps({ horizontal: true });
    expect(wrapper).toMatchSnapshot();
  });
  test("renders end collapsed", () => {
    wrapper.setProps({ collapsed: true, horizontal: false });
    expect(wrapper).toMatchSnapshot();
  });
  test("handleClick is called", () => {
    wrapper.find("CommandBarButton").simulate("click");
    expect(spy).toHaveBeenCalledWith("end", true);
  });
});
