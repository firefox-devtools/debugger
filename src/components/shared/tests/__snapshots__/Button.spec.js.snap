// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders 1`] = `
<div
  className="close-btn error"
  onClick={[Function]}
  title="Close something"
>
  <img
    className="close"
  />
</div>
`;

exports[`CommandBarButton renders 1`] = `
<button
  aria-pressed={false}
  className="command-bar-button resume"
>
  <img />
</button>
`;

exports[`PaneToggleButton renders end 1`] = `
<CommandBarButton
  className="toggle-button end collapsed vertical"
  onClick={[Function]}
  title="Collapse panes"
>
  <Svg
    name="togglePanes"
  />
</CommandBarButton>
`;

exports[`PaneToggleButton renders end collapsed 1`] = `
<CommandBarButton
  className="toggle-button end collapsed vertical"
  onClick={[Function]}
  title="Collapse panes"
>
  <Svg
    name="togglePanes"
  />
</CommandBarButton>
`;

exports[`PaneToggleButton renders end horizontal 1`] = `
<CommandBarButton
  className="toggle-button end collapsed"
  onClick={[Function]}
  title="Collapse panes"
>
  <Svg
    name="togglePanes"
  />
</CommandBarButton>
`;

exports[`PaneToggleButton renders start 1`] = `
<CommandBarButton
  className="toggle-button start vertical"
  onClick={[Function]}
  title="Expand panes"
>
  <Svg
    name="togglePanes"
  />
</CommandBarButton>
`;

exports[`PaneToggleButton renders start collapsed 1`] = `
<CommandBarButton
  className="toggle-button start collapsed vertical"
  onClick={[Function]}
  title="Collapse panes"
>
  <Svg
    name="togglePanes"
  />
</CommandBarButton>
`;

exports[`PaneToggleButton renders start horizontal 1`] = `
<CommandBarButton
  className="toggle-button start"
  onClick={[Function]}
  title="Expand panes"
>
  <Svg
    name="togglePanes"
  />
</CommandBarButton>
`;
